Sources/TootSDK/TootClient/TootClient.swift

```diff
@@ -458,11 +463,38 @@ extension TootClient {
 extension TootClient {
     /// Uses the currently available credentials to connect to an instance and detect the most compatible server flavour.
     public func connect() async throws {
-        if let flavour = await flavourFromNodeInfo() {
-            self.flavour = flavour
+        if let nodeInfo = await getNodeInfoIfAvailable() {
+            self.flavour = nodeInfo.flavour
+            self.versionString = nodeInfo.software.version
+            self.version = TootFeature.parseVersion(from: nodeInfo.software.version)
+            if debugInstance {
+                print("🎨 Detected fediverse instance flavour: \(nodeInfo.flavour), version: \(nodeInfo.software.version)")
+            }
         } else {
-            self.flavour = try await flavourFromInstanceInfo()
+            let instance = try await getInstanceInfo()
+            self.flavour = instance.flavour
+            self.versionString = instance.version
+            self.version = TootFeature.parseVersion(from: instance.version)
+
+            // Store API versions if this is an InstanceV2
+            if let instanceV2 = instance as? InstanceV2 {
+                self.apiVersions = instanceV2.apiVersions
+                if debugInstance, let apiVersions = instanceV2.apiVersions {
+                    print("🎨 Detected API versions - Mastodon: \(apiVersions.mastodon ?? 0)")
+                }
+            }
+
+            if debugInstance {
+                print("🎨 Detected fediverse instance flavour: \(instance.flavour), version: \(instance.version)")
+            }
         }
+
+        // Set the encoder userInfo after flavour has been determined
+        encoder.userInfo[.tootSDKFlavour] = flavour
```

Comment:
I placed that in didSet of a flavour because it is public and can be modified by users of TootSDK. I do that myself when restoring from cache. Moving it here will break cases where someone overrides flavor manually. Overriding flavour/versions could be disallowed and users should be required to call connect or the old placement should be preserved.

I'm myself not yet sure how to update my caching approach to now work with version checks. I added it to avoid api calls since flavour never changed for a given instance.

Reply:
@Tunous That is a good point. I think the "server flavour" is now a combination of flavour and version info. I can still see cases where caching a TootClient may be useful, perhaps it's best to provide a dedicated initialiser for values that can be restored and make the actual properties get only.